scripts/synapse_sytest.sh

```diff
@@ -114,34 +114,32 @@ fi
 # default value for SYNAPSE_SOURCE
 : ${SYNAPSE_SOURCE:=/src}
 
-# if we're running against a source directory, turn it into a tarball.  pip
-# will then unpack it to a temporary location, and build it.  (As of pip 20.1,
-# it will otherwise try to build it in-tree, which means writing changes to the
-# source volume outside the container.)
-#
-if [ -d "$SYNAPSE_SOURCE" ]; then
-    echo "Creating tarball from synapse source"
-    tar -C "$SYNAPSE_SOURCE" -czf /tmp/synapse.tar.gz \
-        synapse setup.py README.rst MANIFEST.in
-    SYNAPSE_SOURCE="/tmp/synapse.tar.gz"
-elif [ ! -r "$SYNAPSE_SOURCE" ]; then
+if [ ! -r "$SYNAPSE_SOURCE" ]; then
     echo "Unable to read synapse source at $SYNAPSE_SOURCE" >&2
     exit 1
 fi
 
+if [ ! -d "$SYNAPSE_SOURCE" ]; then
+    echo "$SYNAPSE_SOURCE must be a source directory" >&2
+    exit 1
+fi
+
+# Make a copy of the source directory to avoid writing changes to the source
+# volume outside the container.
```

Comment:
To check: we're not using this yet, because we're not yet using poetry?

Reply:
We might be using this already: this is another way to avoid writing to the host's source code directory, see the deleted comment on line -117.